decision/peer_request_queue.go

```diff
@@ -195,7 +222,16 @@ func (t *peerRequestTask) SetIndex(i int) {
 }
 
 // taskKey returns a key that uniquely identifies a task.
-func taskKey(p peer.ID, k *cid.Cid) string {
+func taskKey(p peer.ID, entries []*wantlist.Entry) string {
+	key := string(p)
+	for _, entry := range entries {
```

Comment:
This is a bit unfortunate as it could grow quite large (*very* large). I wonder if we can switch to a different method for computing task keys.

Reply:
@Stebalien  Sorry for the delayed response. 
This is done. I pushed out this change and squashed into a single commit.